pkg/server/server.go

```diff
@@ -110,7 +111,17 @@ const clusterWireguardFile = "cluster-wireguard.json"
 
 func WithJSONDB(dataDir, filename string) MapDbOpt {
 	return func(db *mapDB) error {
-		if err := os.MkdirAll(dataDir, 0755); err != nil {
+		bytes, err := os.ReadFile(filepath.Join(dataDir, filename))
+		if err == nil {
+			err := json.Unmarshal(bytes, &(db).db)
+			if err != nil {
+				return err
+			}
+		} else if errors.Is(err, os.ErrNotExist) {
+			if err := os.MkdirAll(dataDir, 0755); err != nil {
+				return err
+			}
+		} else {
```

Comment:
[nit]: We could do an early return to break up how many if branches we have and reduce the cognitive load.